pkg/client/ssa_client.go

```diff
@@ -24,21 +29,59 @@ type SSAApplyClient struct {
 
 	// The field owner to use for SSA-applied objects.
 	FieldOwner string
+
+	// MigrateSSAByDefault specifies the default SSA migration behavior.
+	//
+	// When checking for the migration, there is an additional GET of the resource, followed by optional
+	// UPDATE (if the migration is needed) before the actual changes to the objects are applied.
+	//
+	// This field specifies the default behavior that can be overridden by supplying an explicit MigrateSSA() option
+	// to ApplyObject or Apply methods.
+	//
+	// The main advantage of using the SSA in our code is that ability of SSA to handle automatic deletion of fields
+	// that we no longer set in our templates. But this only works when the fields are owned by managers and applied
+	// using "Apply" operation. As long as there is an "Update" entry with given field (even if the owner is the same)
+	// the field WILL NOT be automatically deleted by Kubernetes.
+	//
+	// Therefore, we need to make sure that our manager uses ONLY the Apply operations. This maximizes the chance
+	// that the object will look the way we need.
+	MigrateSSAByDefault bool
+
+	// NonSSAFieldOwner should be set to the same value as the user agent used by the provided Kubernetes client
+	// or to the value of the explicit field owner that the calling code used to use with the normal CRUD operations
+	// (highly unlikely and not the case in our codebase).
+	//
+	// The user agent can be obtained from the REST config from which the client is constructed.
+	//
+	// The user agent in the REST config is usually empty, so there's no need to set it here either in that case.
+	NonSSAFieldOwner string
```

Comment:
in your code in the previous PR, you originally had a code that was able to construct the owner name we should migrate from - couldn't we use that?
https://github.com/metlos/toolchain-common/blob/6ff87dfbb853a19475dff63b22a0e5a2f51300d9/pkg/client/ssa_client.go#L50-L62

Reply:
I wanted to have this option available though in case I missed something and we for example set the user agent of the kubernetes client to an explicit value somewhere in the code.

In that case, we'd need to set this to an explicit value, too.